src/shared/bus-unit-util.c

```diff
@@ -3138,3 +3140,91 @@ ExecDirectoryFlags exec_directory_flags_from_string(const char *s) {
 
         return _EXEC_DIRECTORY_FLAGS_INVALID;
 }
+
+static int read_wakeup_count(uint64_t *ret_wakeup_count) {
+        _cleanup_free_ char *t = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/power/wakeup_count", &t);
```

Comment:
Well, this one seems problematic.

First of all, UnitFreezer isn't only about sleeping. homed's lock operation employs this too, and that can happen outside of sleep context. Plus, this adds IMO unnecessary churn. Maybe it's better to leave the wakeup management completely to kernel...

Reply:
Dropped for now. If there's a need this can be reintroduced later.

Reply:
It's also worth pointing out that this is implemented in a new function that's not used in `systemd-homed`, only `systemd-sleep`.